mobile_config/src/gateway_info_v3.rs

```diff
@@ -0,0 +1,282 @@
+use chrono::{DateTime, Utc};
+use helium_crypto::PublicKeyBinary;
+use helium_proto::services::mobile_config::{
+    DeploymentInfo as DeploymentInfoProto, DeviceTypeV2 as DeviceTypeProtoV2,
+    GatewayInfoV3 as GatewayInfoProtoV3, GatewayMetadataV3 as GatewayMetadataProtoV3,
+    LocationInfo as LocationInfoProto,
+};
+
+use crate::gateway_info::DeviceTypeParseError;
+
+#[derive(Clone, Debug)]
+pub struct LocationInfo {
+    pub location: u64,
+    pub location_changed_at: DateTime<Utc>,
+}
+
+#[derive(Clone, Debug)]
+pub struct GatewayMetadataV3 {
+    pub location_info: LocationInfo,
+    pub deployment_info: Option<DeploymentInfoProto>,
+}
+
+#[derive(Clone, Debug)]
+pub enum DeviceTypeV2 {
+    Indoor,
+    Outdoor,
+    DataOnly,
+}
+
+impl std::str::FromStr for DeviceTypeV2 {
+    type Err = DeviceTypeParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let result = match s {
+            "wifiIndoor" => Self::Indoor,
+            "wifiOutdoor" => Self::Outdoor,
+            "wifiDataOnly" => Self::DataOnly,
+            _ => return Err(DeviceTypeParseError),
+        };
+        Ok(result)
+    }
+}
+
+impl DeviceTypeV2 {
+    fn to_sql_param(&self) -> &'static str {
+        match self {
+            DeviceTypeV2::Indoor => "wifiIndoor",
+            DeviceTypeV2::Outdoor => "wifiOutdoor",
+            DeviceTypeV2::DataOnly => "wifiDataOnly",
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct GatewayInfoV3 {
+    pub address: PublicKeyBinary,
+    pub metadata: Option<GatewayMetadataV3>,
```

Comment:
Is there a way receive a `GatewayInfoV3` without metadata being present?

Reply:
Yes, there is. Added `gateway_stream_info_v3_no_metadata`  test